sympy/functions/combinatorial/numbers.py

```diff
@@ -1861,9 +1861,19 @@ def stirling(n, k, d=None, kind=2, signed=False):
 def _nT(n, k):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if k > n or k == 0:
+        return 0
+    if n == k or k == 1:
```

Comment:
I believe these two lines should appear first to handle the `n==k==0` case (in case this is called directly). A test to add to the `test_numbers.py` file in the function that tests `partition` would be
```python
assert [_nT(i,j) for i in range(5) for j in range(i + 2)] == [1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 1, 2, 1, 1, 0]
```
(That should fail in your current PR but pass once those two lines are move.)

I don't think a reference is needed after all since this is just the dynamic programming approach for this problem and is given [here](https://codereview.stackexchange.com/questions/171551/finding-number-of-integer-partitions), too.